app/models/locations/location.rb

```diff
@@ -60,6 +61,11 @@ def parent
     super || NullLocation.new
   end
 
+  def parent=(new_parent)
+    super
+    self.internal_parent = new_parent
+  end
+
```

Comment:
The before save handles most of this, but this makes sure things like factories continue to behave properly. (I'd get rid of the before save, but that will catch some cases this might miss, like setting parent by id)